protocol-impl/src/main/java/io/zeebe/protocol/impl/record/value/processinstance/ProcessEventRecord.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.zeebe.protocol.impl.record.value.processinstance;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.zeebe.msgpack.property.DocumentProperty;
+import io.zeebe.msgpack.property.LongProperty;
+import io.zeebe.msgpack.property.StringProperty;
+import io.zeebe.protocol.impl.encoding.MsgPackConverter;
+import io.zeebe.protocol.impl.record.UnifiedRecordValue;
+import io.zeebe.protocol.record.value.ProcessEventRecordValue;
+import io.zeebe.util.buffer.BufferUtil;
+import java.util.Map;
+import org.agrona.DirectBuffer;
+
+@SuppressWarnings("java:S2160")
+public final class ProcessEventRecord extends UnifiedRecordValue
```

Comment:
(optional) We could add two additional properties `processInstanceKey` and `processDefinitionKey` to provide more context for the event. These properties are not required for the execution but they could be handy for testing/debugging and for a consumer of the stream.

We have these properties also for variable records.

Reply:
I was on the fence, as it's easy to add properties, but harder to remove them in a backwards compatibility way - but I see no harm, and even though we don't have a specific consumer for these, we've learned that they're generally useful to scope records. Let's do it :+1: